nncf/torch/knowledge_distillation_loss.py

```diff
@@ -0,0 +1,151 @@
+from copy import deepcopy
+from functools import reduce, partial
+
+from torch import nn
+import torch
+
+from nncf.common.schedulers import BaseCompressionScheduler
+from nncf.common.statistics import NNCFStatistics
+from nncf.torch.graph.transformations.layout import PTTransformationLayout
+from nncf import NNCFConfig
+from nncf.torch.nncf_network import NNCFNetwork
+from nncf.torch.compression_method_api import PTCompressionAlgorithmBuilder
+from nncf.torch.compression_method_api import PTCompressionAlgorithmController
+from nncf.api.compression import CompressionLoss, CompressionScheduler, CompressionStage
+from nncf.torch.algo_selector import COMPRESSION_ALGORITHMS
+from nncf.torch.compression_method_api import PTCompressionLoss
+from nncf.torch.nested_objects_traversal import nested_object_paths_generator
+
+
+class KnowledgeDistillationLoss(PTCompressionLoss):
```

Comment:
Should state explicitly somewhere in a docstring that only the inputs of the model that were consumed by the model during the latest `forward` call will be used to determine original model output to be compared with the compressed model's current output. This is a limitation of the algorithm, so we need to make sure that the user understands it.